pkg/controllers/state/metrics.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	crmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	"sigs.k8s.io/karpenter/pkg/metrics"
+)
+
+const (
+	stateSubsystem = "state"
+)
+
+var (
+	ClusterStateNodesTotal = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: metrics.Namespace,
+			Subsystem: stateSubsystem,
+			Name:      "nodes_total",
```

Comment:
```suggestion
			Name:      "nodes_count",
```

Looking at upstream K8s, "total" is always in reference to a counter metric because it's the total over the course of the process. I was original thinking that, since this was a gauge, this should just be "nodes" like `kapenter_state_nodes` but I could also see this just being `karpenter_state_nodes_count` to make it more clear that this is the current count of nodes right now

Reply:
I like that, accepted